python/mxnet/gluon/metric.py

```diff
@@ -617,92 +585,102 @@ def update_binary_stats(self, label, pred):
         pred : `NDArray`
             Predicted values.
         """
-        pred = pred.asnumpy()
-        label = label.asnumpy().astype('int32')
-        pred_label = numpy.argmax(pred, axis=1)
-
-        check_label_shapes(label, pred)
-        if len(numpy.unique(label)) > 2:
-            raise ValueError("%s currently only supports binary classification."
-                             % self.__class__.__name__)
+        pred = pred.as_np_ndarray()
+        label = label.as_np_ndarray().astype('int32')
+        if self.class_type == "binary":
+            self._set(1)
+            if len(numpy.unique(label)) > 2:
```

Comment:
This will trigger synchronization (as we need to wait for the result of the np.unique operator). Could we make error checking that triggers synchronization optional?

Reply:
https://github.com/apache/incubator-mxnet/blob/83b51703ed354f41024423f140de38df2ba22d50/src/imperative/imperative.cc#L123-L127